packages/components/pie-webc-core/src/test/forms/pie-form-manager.spec.ts

```diff
@@ -0,0 +1,191 @@
+import { PieFormManager } from '../../forms/pie-form-manager';
+
+describe('PieFormManager', () => {
+    describe('addForm', () => {
+        it('correctly sets up a submit event listener when a form is added', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+            const addEventListenerSpy = vi.spyOn(testForm, 'addEventListener');
+
+            // Act
+            manager.addForm(testForm);
+
+            // Assert
+            expect(addEventListenerSpy).toHaveBeenCalledWith('submit', expect.any(Function));
+        });
+
+        it('does not throw an error when trying to add the same form multiple times', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+
+            // Act
+            const addForm = () => manager.addForm(testForm);
+
+            // Assert
+            expect(addForm).not.toThrow();
+        });
+    });
+
+    describe('getForm', () => {
+        it('returns the PieFormData object associated with a form', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+            manager.addForm(testForm);
+
+            // Act
+            const result = manager.getForm(testForm);
+
+            // Assert
+            expect(result?.form).toStrictEqual(testForm);
```

Comment:
Do we also want the listener, for completeness?

Reply:
![for-completeness](https://github.com/justeattakeaway/pie/assets/16766185/d13ca7e7-e24d-492a-9da2-4c8db95847b3)